benchmark/src/main/scala/mesosphere/marathon/core/storage/zookeeper/ZooKeeperPersistenceStoreBenchmark.scala

```diff
@@ -0,0 +1,166 @@
+
+  lazy val settings: AsyncCuratorBuilderSettings = new AsyncCuratorBuilderSettings(createOptions = Set(CreateOption.createParentsIfNeeded), compressedData = false)
+  lazy val factory: AsyncCuratorBuilderFactory = AsyncCuratorBuilderFactory(curator, settings)
+  lazy val store: ZooKeeperPersistenceStore = new ZooKeeperPersistenceStore(factory, parallelism = 16)
+
+  // An map of node size to number of nodes of that size. Used for read, update and delete benchmarks. Note that
+  // different number of nodes is used depending on the node data size e.g. creating 10K nodes with 1Mb data is not
+  // practical since it will result in 10Gb of data.
+  type NodeSize = Int
+  type NodeNumber = Int
+  val params = Map[NodeSize, NodeNumber]((10, 10000), (100, 10000), (1024, 10000), (10240, 10000), (102400, 1000))
+
+  /**
+    * Helper method to pre-populate Zookeeper with data. By default nodes are created with path:
+    * /tests/{size}/node{index} e.g. "/tests/100/node123" for a 123th node with data size 100b.
+    *
+    * @param args
+    */
+  def main(args: Array[String]): Unit = {
+    def populate(size: Int, num: Int) = {
+      Source(1 to num)
+        .map(i => Node(s"/tests/$size/node$i", ByteString(Random.alphanumeric.take(size).mkString)))
+        .via(store.createFlow)
+        .runWith(Sink.ignore)
+    }
+
+    Await.result(
+      Source.fromIterator(() => params.iterator)
+        .map{ p => logger.info(s"Creating ${p._2} nodes with ${p._1}b data"); p }
+        .map{ case (size, num) => populate(size, num) }
+        .mapAsync(1)(identity)
+        .runWith(Sink.ignore),
+      Duration.Inf)
+
+    logger.info("Zookeeper successfully populated with data")
+    system.terminate()
+  }
+}
+
+@Fork(1)
+@State(Scope.Benchmark)
+@OutputTimeUnit(TimeUnit.SECONDS)
+@BenchmarkMode(Array(Mode.Throughput))
+@Warmup(iterations = 15, time = 1, timeUnit = TimeUnit.SECONDS)
+@Measurement(iterations = 5, time = 1, timeUnit = TimeUnit.SECONDS)
+class ZooKeeperPersistenceStoreBenchmark {
+  import ZooKeeperPersistenceStoreBenchmark._
+
+  /** Node data size */
+  @Param(value = Array("10", "100", "1024", "10240", "102400"))
+  var size: Int = _
+
+  /** Number of nodes per operation (Source size) */
+  @Param(value = Array("1", "10", "100", "1000"))
+  var num: Int = _
+
+  def randomPath(prefix: String = "", size: Int = 10): String =
+    s"$prefix/${Random.alphanumeric.take(size).mkString}"
+
+  @Benchmark
+  def creates(hole: Blackhole) = {
+    // For created nodes not to collide with each other we use random node names generated by `randomPath`
+    val res = Await.result(
+      Source(1 to num)
+        .map(_ => Node(randomPath("/tests"), ByteString(Random.alphanumeric.take(size).mkString)))
+        .via(store.createFlow)
+        .runWith(Sink.ignore), Duration.Inf)
+  }
+
+  //  @Benchmark
+  def reads(hole: Blackhole) = {
+    def paths = Source(1 to num)
+      .map(_ => s"/tests/$size/node${Random.nextInt(params(size)) + 1}")
```

Comment:
`val params = Map[NodeSize, NodeNumber]((10, 10000), ...` defined above. It stores how many elements of that size where created in the `main` method. This is basically an additional mapping between `@Param size` and  `@Param num`. But because JMH treats both `@Param`s independently from each other, I needed this ugly workaround.